Kickstarter-iOS/Views/Controllers/LoginViewController.swift

```diff
@@ -132,9 +132,19 @@ internal final class LoginViewController: UIViewController {
       }
 
     self.viewModel.outputs.logIntoEnvironment
+      .observeForControllerAction()
       .observeValues { [weak self] env in
+        guard let self = self else { return }
         AppEnvironment.login(env)
-        self?.viewModel.inputs.environmentLoggedIn()
+
+        if featureEmailVerificationFlowIsEnabled() {
```

Comment:
Any control flow logic like this should as far as possible live in the view model so that we can test it.